BlockServer/alarm/load_alarm_config.py

```diff
@@ -0,0 +1,58 @@
+import subprocess
+import threading
+import os
+import time
+
+
+class AlarmConfigLoader(object):
+
+    # Instance of this singleton
+    instance = None
+    # The subprocess that the alarm server reloads itself in
+    process = None
+    # Number of seconds to delay the reload by
+    # (the blockserver must have finished loading the configuration before the alarm server can be restarted)
+    delay = 20
+
+    @staticmethod
+    def get_instance():
+
+        if AlarmConfigLoader.instance is None:
+            AlarmConfigLoader.instance = AlarmConfigLoader()
+
+        return AlarmConfigLoader.instance
+
+    def _load(self):
+        """
+        Simple wrapper around a batch script which reloads the configuration in the alarm server.
+        """
+
+        # Terminate the last batch job if it is still running
+        # (prevents two instances from running concurrently which might cause problems).
+        if self.process is not None:
```

Comment:
In cases where you use static (i.e. class) variables in Python, it is better to use the class name rather than self (e.g. http://python-3-patterns-idioms-test.readthedocs.io/en/latest/Singleton.html). Although `self` works, it gives the impression the variable belongs to a particular instance of the object rather than being a class variable. If you edit self.class_variable, it'll affect every other instance of the class so the side-effects can be unexpected. Although Singletons only ever have one instance of the class, I feel it would be better to use the standard notation.